File: lib_ebook_db/src/main/java/com/ebook/db/entity/SearchHistory.java
package com.ebook.db.entity;

import org.greenrobot.greendao.annotation.Entity;
import org.greenrobot.greendao.annotation.Generated;
import org.greenrobot.greendao.annotation.Id;

@Entity
public class SearchHistory {
    @Id(autoincrement = true)
    private Long id = null;
    private int type;
    private String content;
    private long date;

    @Generated(hash = 1127448272)
    public SearchHistory(Long id, int type, String content, long date) {
        this.id = id;
        this.type = type;
        this.content = content;
        this.date = date;
    }

    @Generated(hash = 1905904755)
    public SearchHistory() {
    }

    public SearchHistory(int type, String content, long date) {
        this.type = type;
        this.content = content;
        this.date = date;
    }

    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getType() {
        return this.type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getContent() {
        return this.content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public long getDate() {
        return this.date;
    }

    public void setDate(long date) {
        this.date = date;
    }
}

File: lib_ebook_db/src/main/java/com/ebook/db/entity/SearchHistory.kt
package com.ebook.db.entity

import android.os.Parcelable
import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity
data class SearchHistory(
    var type: Int = 0,
    @JvmField
    var content: String = String(),
    @JvmField
    var date: Long = 0,
    @Id var id: Long = 0
) : Parcelable {
    constructor(type: Int, content: String, data: Long) : this(
        type = type,
        content = content,
        date = data
    )
}